### PR TITLE
feat(application-system): Unlist applications for certain role/state combinations

### PR DESCRIPTION
**Pull request by Thorrik58**
# Unlist applications for certain role/state combinations

Attach a link to issue if relevant

## What

Unlist applications for certain role/state combinations

## Why

Needed in order to hide the applications for certain reviewers.

## Screenshots / Gifs

Attach Screenshots / Gifs to help reviewers understand the scope of the pull request

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

File: apps/application-system/api/src/app/modules/application/e2e/applicationAccess.spec.ts [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+    const results2 = await applicationAccessService.shouldShowApplicationOnOverview(
+      applicationInDraft,
+      '111111-3001',
+      testApplicationTemplate,
+    )
+    expect(results2).toBe(true)
+  })
+
+  it('should return correct ', async () => {
```
Comment (saevarma) [inline apps/application-system/api/src/app/modules/application/e2e/applicationAccess.spec.ts:75]: Should return correct for what role/state combination?

File: apps/application-system/api/src/app/modules/application/tools/applicationAccess.service.ts [hunk 2]
```diff
@@ -8,8 +8,15 @@ import {
...
-import { Application, ApplicationTypes } from '@island.is/application/types'
+import {
+  Application,
+  ApplicationContext,
+  ApplicationStateSchema,
+  ApplicationTemplate,
+  ApplicationTypes,
+} from '@island.is/application/types'
 import { getApplicationTemplateByTypeId } from '@island.is/application/template-loader'
+import { EventObject } from 'xstate'
```
Comment (saevarma) [inline apps/application-system/api/src/app/modules/application/tools/applicationAccess.service.ts:19]: Does this pass format with no line between imports and implementation?

Comment (Thorrik58) [reply apps/application-system/api/src/app/modules/application/tools/applicationAccess.service.ts:19]: Formatted before pushing so it should

File: apps/application-system/api/src/app/modules/application/e2e/applicationAccess.spec.ts (hunk 1 above)
Comment (Thorrik58) [reply apps/application-system/api/src/app/modules/application/e2e/applicationAccess.spec.ts]: Removed and made clearer

File: apps/application-system/api/src/app/modules/application/e2e/applicationAccess.spec.ts [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+      applicationInReview,
+      '111111-3000',
+    )
+    expect(results).toBe(false)
+  })
+
+  it('should return true when application is in draft and has both shouldBeListedForRole set as true for applicant and undefined for reviewer', async () => {
+    const applicationInDraft = createApplication({
+      state: 'draft',
+      applicant: '111111-3000',
```
[the remark this answers is not in the document]
Comment (Thorrik58) [reply apps/application-system/api/src/app/modules/application/e2e/applicationAccess.spec.ts:58]: Nope just for this test template. 
Draft applicant - shouldBeListedForRole = true
Draft reviewer - shouldBeListedForRole = undefined
InReview applicant - shouldBeListedForRole = undefined
InReview reviewer - shouldBeListedForRole = false

If undefined we want to show the application for the role as every application except this one doesn't have it defined.

File: apps/application-system/api/src/app/modules/application/e2e/applicationAccess.spec.ts [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+    })
+    const results = await applicationAccessService.canDeleteApplication(
+      applicationInReview,
+      '111111-3000',
+    )
+    expect(results).toBe(false)
+  })
+
+  // true check for applicant in draft defined as true
```
Comment (saevarma) [inline apps/application-system/api/src/app/modules/application/e2e/applicationAccess.spec.ts:54]: Unnecessary  comment, the test description describes itself.

File: apps/application-system/api/src/app/modules/application/e2e/applicationAccess.spec.ts [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+    })
+    const results = await applicationAccessService.shouldShowApplicationOnOverview(
+      applicationInReview,
+      '111111-3001',
+      testApplicationTemplate,
+    )
+    expect(results).toBe(false)
+  })
+
+  it('should return true when application is in review and user is applicant, shouldBeListed defined as true', async () => {
```
Comment (saevarma) [inline apps/application-system/api/src/app/modules/application/e2e/applicationAccess.spec.ts:96]: `shouldBeListedForRole` is undefined for applicant in the `inReview` state.

Just FYI: this is a bit fragile tests, as the `createApplicationTemplate` is a common tool in our shared testing fixtures library. Anyone could change the role setup later on and the "default" values of the `shouldBeListedForRole`, possible breaking these tests.

**Review by saevarma — APPROVED** [review]
Just two minor comments that you can review if you want to act on.
Otherwise LGTM.

**Review by obmagnusson — APPROVED** [review]
LGTM

File: apps/application-system/api/src/app/modules/application/e2e/applicationAccess.spec.ts (hunk 5 above)
Comment (Thorrik58) [reply apps/application-system/api/src/app/modules/application/e2e/applicationAccess.spec.ts:96]: True but thats the case for 3 of the tests, ill look into changing the test.

File: apps/application-system/api/src/app/modules/application/e2e/applicationAccess.spec.ts (hunk 4 above)
Comment (Thorrik58) [inline apps/application-system/api/src/app/modules/application/e2e/applicationAccess.spec.ts:54]:
```suggestion
```

**Pull request merged**